core/src/main/java/hudson/model/Fingerprint.java

```diff
@@ -1234,74 +1239,34 @@ public int compare(FingerprintFacet o1, FingerprintFacet o2) {
     }
 
     /**
-     * Save the settings to a file.
+     * Save the Fingerprint in the Fingerprint Storage
      * @throws IOException Save error
      */
     public synchronized void save() throws IOException {
-        if(BulkChange.contains(this))   return;
+        if(BulkChange.contains(this)) {
+            return;
+        }
 
-        long start=0;
+        long start = 0;
         if(logger.isLoggable(Level.FINE))
             start = System.currentTimeMillis();
 
-        File file = getFingerprintFile(md5sum);
-        save(file);
-        SaveableListener.fireOnChange(this, getConfigFile(file));
+        // Implementations are expected to invoke SaveableListener on their own if relevant
+        // TODO: Consider improving Saveable Listener API: https://issues.jenkins-ci.org/browse/JENKINS-62543
+        FingerprintStorage.get().save(this);
 
         if(logger.isLoggable(Level.FINE))
-            logger.fine("Saving fingerprint "+file+" took "+(System.currentTimeMillis()-start)+"ms");
+            logger.fine("Saving fingerprint " + getHashString() + " took " + (System.currentTimeMillis() - start) + "ms");
```

Comment:
Curlies around this if please

Reply:
@afalko thanks for reviewing :)
I want to avoid this because of https://github.com/jenkinsci/jenkins/pull/4731#discussion_r437368226. 
But I am happy to add them if you feel

Reply:
So long as you do it a follow up PR, I'm fine with it